Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs

```diff
@@ -134,6 +134,14 @@ public async ValueTask DeleteAsync(string id, PartitionKey partitionKey, Cancell
             Items.TryRemove(item!.Id, out _);
         }
 
+        /// <inheritdoc/>
+        public async ValueTask<int> CountAsync(Expression<Func<TItem, bool>> predicate,
```

Comment:
There are some unit tests for this implementation `InMemoryRepostitoryTests.cs` adding some cases for these would be great.